lib/src/blaze/fetch_full_transaction.rs

```diff
@@ -163,6 +169,52 @@ impl TransactionContext {
         //info!("Finished Fetching full transaction {}", tx.txid());
     }
 
+    async fn update_outgoing_metadatas_with_uas(
+        &self,
+        arbitrary_memos_with_txids: Vec<([u8; 511], TxId)>,
+    ) {
+        for (wallet_internal_data, txid) in arbitrary_memos_with_txids {
+            let mut uas_bytes = vec![0; 511];
+            wallet_internal_data
+                .as_slice()
+                .read(&mut uas_bytes)
+                .unwrap();
+            let uas_string = String::from_utf8(uas_bytes);
```

Comment:
We should be using this encoding:  https://zips.z.cash/zip-0316#encoding-of-unified-addresses

Reply:
Unless there's a compression of that?